backend/api/professor/invitations.py

```
@@ -414,3 +430,573 @@ async def get_cohort_enrollment_stats(
+    # Cache the result until expiry (with a small buffer)
+    ttl = int((invite.expires_at - datetime.now(timezone.utc)).total_seconds())
+    if ttl > 0:
+        redis_manager.set(cache_key, response_data, min(ttl, 3600))  # Cache for up to 1 hour
+    
+    return response_data
+
+
+@public_router.post("/invites/{token}/accept")
+async def accept_invite_link(
+    token: str,
+    current_user: Optional[User] = Depends(get_current_user_optional),
```

Ruff (0.14.3)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)